components/EnhancedDataTable.tsx

```diff
@@ -182,7 +184,7 @@ export default class EnhancedDataTable<T = any> extends React.Component<
     get columns(): IDataTableColumn[] {
         return (this.props.columns || []).map((c) => ({
             ...c,
-            omit: c.name ? !this.columnVisibility[c.name.toString()] : c.omit,
+            omit: c.name ? !this.columnVisibility[getColumnKey(c)] : c.omit,
```

Comment:
In case `name` is a component `toString` does not really work. We should use the column key instead.